pkg/virt-handler/vm.go

```diff
@@ -1119,6 +1119,15 @@ func (d *VirtualMachineController) updateVMIStatus(origVMI *v1.VirtualMachineIns
 		log.Log.Errorf("virt-launcher does not support the Secure Boot setting. Updating VMI %s status to Failed", vmi.Name)
 		vmi.Status.Phase = v1.Failed
 	}
+
+	if vmi.ResourceVersion == "" {
+		vmiCopy, err := d.clientset.VirtualMachineInstance(vmi.ObjectMeta.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+		if err != nil {
+			log.Log.Object(vmi).Infof("Can not get resourceVersion for Updating VMI %s status to Failed, error %v", vmi.Name, err.Error())
+			return err
+		}
+		vmi.ResourceVersion = vmiCopy.ResourceVersion
```

Comment:
Would be good to func test this as well.

Reply:
ye, I added a functest for virt-handler getting latest resourceVersion for updating vmi